service-base/src/main/java/org/eclipse/hono/service/metric/MicrometerBasedMetrics.java

```diff
@@ -30,9 +32,13 @@
 import io.micrometer.core.instrument.Tags;
 import io.micrometer.core.instrument.Timer;
 import io.micrometer.core.instrument.Timer.Sample;
+import io.vertx.core.Vertx;
 
 /**
- * Micrometer based metrics implementation.
+ * Micrometer based metrics implementation. Checks if tenant {@link ProtocolAdapterProperties#getIdleTenantTimeout()} is
+ * exceeded and publishes the tenantId on the event bus address {@link Constants#EVENT_BUS_ADDRESS_TENANT_TIMED_OUT}
+ * once a tenant times out.
```

Comment:
Speaking of a "tenant time out" might be confusing.
How about "once the tenant idle timeout has been reached"?
And maybe use `<p>Also checks if [...]` above, making clear the idle timeout check is not the main purpose of the class.